File: app/src/main/java/com/txt_nifty/sketch/flmml/MMLEvent.java
package com.txt_nifty.sketch.flmml;

public class MMLEvent {
    public static final String SIGNAL = "signal";
    public static final String COMPLETE = "complete";
    public static final String COMPILE_COMPLETE = "compileComplete";
    public static final String BUFFERING = "buffering";
    public long globalTick;
    public int id;
    public int progress;
    public String type;

    public MMLEvent(String aType) {
        this(aType, 0, 0, 0);
    }

    public MMLEvent(String aType, long aGlobalTick, int aId, int aProgress) {
        type = aType;
        globalTick = aGlobalTick;
        id = aId;
        progress = aProgress;
    }

}

File: app/src/main/java/com/txt_nifty/sketch/flmml/MMLEvent.kt
package com.txt_nifty.sketch.flmml

class MMLEvent constructor(
    val type: String,
    val globalTick: Long = 0,
    val id: Int = 0,
    val progress: Int = 0
) {
    companion object {
        const val SIGNAL = "signal"
        const val COMPLETE = "complete"
        const val COMPILE_COMPLETE = "compileComplete"
        const val BUFFERING = "buffering"
    }
}